ui/components/app/signature-request/signature-request.js

```diff
@@ -0,0 +1,403 @@
+import React, { useContext, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { memoize } from 'lodash';
+
+import {
+  accountsWithSendEtherInfoSelector,
+  conversionRateSelector,
+  doesAddressRequireLedgerHidConnection,
+  getCurrentCurrency,
+  getPreferences,
+  getSelectedAccount,
+  getSubjectMetadata,
+  getTotalUnapprovedMessagesCount,
+  unconfirmedMessagesHashSelector,
+} from '../../../selectors';
+import {
+  getNativeCurrency,
+  getProviderConfig,
+  isAddressLedger,
+} from '../../../ducks/metamask/metamask';
+import {
+  getAccountByAddress,
+  getURLHostName,
+  sanitizeMessage,
+  valuesFor,
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  shortenAddress,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../helpers/utils/util';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { ConfirmPageContainerNavigation } from '../confirm-page-container';
+import NetworkAccountBalanceHeader from '../network-account-balance-header';
+import SecurityProviderBannerMessage from '../security-provider-banner-message';
+import SiteOrigin from '../../ui/site-origin';
+import LedgerInstructionField from '../ledger-instruction-field';
+import ContractDetailsModal from '../modals/contract-details-modal';
+import { clearConfirmTransaction } from '../../../ducks/confirm-transaction/confirm-transaction.duck';
+import { cancelMsgs, showModal } from '../../../store/actions';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { formatCurrency } from '../../../helpers/utils/confirm-tx.util';
+import { getValueFromWeiHex } from '../../../../shared/modules/conversion.utils';
+import { Numeric } from '../../../../shared/modules/Numeric';
+import { EtherDenomination } from '../../../../shared/constants/common';
+import { MetaMetricsEventCategory } from '../../../../shared/constants/metametrics';
+import { MESSAGE_TYPE } from '../../../../shared/constants/app';
+import { SECURITY_PROVIDER_MESSAGE_SEVERITIES } from '../security-provider-banner-message/security-provider-banner-message.constants';
+import { NETWORK_TYPES } from '../../../../shared/constants/network';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+
+import Typography from '../../ui/typography';
+import Button from '../../ui/button';
```

Comment:
You could update this file to use the `Text` and `Button` components from the `component-library` as `Typography` and this version of the `Button` is deprecated

Reply:
Thanks for the input @georgewrmarshall , I will update it now